proposed/2021/NuGet-org-login-auth-for-dotnet-nuget-push.md

```diff
@@ -0,0 +1,79 @@
+# NuGet.org login authentication workflow for `dotnet nuget push`
+
+- Author: Christopher Gill (https://github.com/chgill-MSFT)
+- Start Date: 2021-03-1
+- GitHub Issue: https://github.com/NuGet/Home/issues/10657
+- GitHub PR: N/A
+
+## Summary
+
+Currently there are 3 ways to push NuGet packages to NuGet.org:
+* [Manual upload on NuGet.org](https://www.nuget.org/packages/manage/upload)
+* `donet nuget push` with a valid API key
+* `nuget push` with a valid API key
+
+I propose creating a workflow to push to NuGet.org via the `dotnet nuget push` command by NuGet.org account login in the CLI or the browser.
+
+## Motivation 
+
+1. API keys can be a pain to use in many circumstances, especially for beginners. To securely upload a package to NuGet.org via the CLI you need to:
+   1. Create a package.
+   2. Login to NuGet.org using MSA.
+   3. Create an API key with the correct permissions/scope. Beginners will be presented with potentially overwhelming options and technical terms like expiration time, scope, unlist, and glob pattern. This is complicated enough, we have an entire [doc on how to do it securely](https://docs.microsoft.com/en-us/nuget/nuget-org/scoped-api-keys).
+   4. Either store the API key securely. Beginners will need to do research on how to do this or may insecurely store the API key in plaintext. The best way to do this in Windows is with `nuget setapikey`, which currently has no support on MacOS or Linux.
+   5. Push the package using the very long `dotnet nuget push <package id> --api-key <api key> --source https://api.nuget.org/v3/index.json` command, which is unlikely to figure out without copying from docs.
```

Comment:
You really shouldn't do this. You're leaking your API key in plaintext. Keep in mind most operating systems store your command history to disk (like the `.bash_history` file).

Reply:
Which part, number 5?

Reply:
Sorry, I meant that you shouldn't avoid providing the `--api-key` through the `dotnet nuget push` command.